For_university/experiment1.py

```diff
@@ -1,71 +1,66 @@
 
 
-def making_universe(SIZE_OF_UNIVERSE, CARDINALITY):
+def make_universe():
     universe = []
-    i = 0
     small_subset = int(SIZE_OF_UNIVERSE / CARDINALITY - 1)
     big_subset = int(SIZE_OF_UNIVERSE / CARDINALITY + 1)
     last_element = 0
     for index in range(CARDINALITY):
         if index < int(CARDINALITY / 2):
             subset = list(ELEMENTS_OF_UNIVERSE[last_element:last_element + big_subset])
-            for i in range(NUMBER_OF_DUPLICATE+1):
+            for i in range(NUMBER_OF_DUPLICATE + 1):
                 universe.append(subset)
             last_element = last_element + big_subset
         else:
             universe.append(list(ELEMENTS_OF_UNIVERSE[last_element:last_element + small_subset]))
             last_element = last_element + small_subset
 
-
     return universe
 
 
-def coverage_function(subset):
-    return len(set(UNCOVERED_ELEMENTS).difference(set(subset)))
+def coverage_routine(subset, uncovered_elements):
+    return len(uncovered_elements.difference(set(subset)))
 
 
-def find_best_in_segment(subsets_in_segment):
-    best_subset = (min(subsets_in_segment, key=coverage_function))
+def find_best_in_segment(subsets_in_segment, uncovered_elements):
+    best_subset = (min(subsets_in_segment, key=lambda subset: coverage_routine(subset, uncovered_elements)))
+    print('This is the best subset' + str(best_subset))
     return best_subset
 
 
-def proposed_algorithem(subsets):
-    global UNCOVERED_ELEMENTS
-    solution = list()
+def proposed_algorithm(subsets):
+    uncovered_elements = set(ELEMENTS_OF_UNIVERSE)
+    solution = []
     for i in range(CARDINALITY):
-        best_marginal_subset = find_best_in_segment(subsets[i * RATIO:i * RATIO + RATIO])
+        best_marginal_subset = find_best_in_segment(subsets[i * RATIO:i * RATIO + RATIO], uncovered_elements)
         solution.append(best_marginal_subset)
-        UNCOVERED_ELEMENTS = UNCOVERED_ELEMENTS.difference(set(best_marginal_subset))
-    return solution
+        uncovered_elements = uncovered_elements - set(best_marginal_subset)
+    return solution, uncovered_elements
 
 
 def main():
-    global UNCOVERED_ELEMENTS
-    subsets = making_universe(SIZE_OF_UNIVERSE, CARDINALITY)
-    print(subsets)
-    results_sum = float(0)
+    print('helllooo')
+    subsets = make_universe()
+    print('This is the universe: ' + str(subsets))
+    results_sum = 0.0
     for i in range(TEST_LOOP):
         subsets = random.sample(subsets, NUMBER_OF_SUBSETS)
-        solution = proposed_algorithem(subsets)
-        results_sum += (1 - float(len(UNCOVERED_ELEMENTS) / SIZE_OF_UNIVERSE))
-        UNCOVERED_ELEMENTS = set(ELEMENTS_OF_UNIVERSE)
-
-    print('The expectation of the competetive ratio is %f' % float(results_sum/TEST_LOOP))
-
-
-
+        print('This is the universe after sampling: ' + str(subsets))
+        solution, uncovered_elements = proposed_algorithm(subsets)
+        results_sum += (1 - float(len(uncovered_elements) / SIZE_OF_UNIVERSE))
```

Comment:
why do you cast to `float`? 
python is pretty smart about choosing numeric types. I think it's redundant.